client/src/pages/GrantAccessForm.js

```diff
@@ -0,0 +1,220 @@
+import { useState } from "react";
+import "./GrantAccessForm.css";
+import { post } from "../TenderClient";
+import Logo from "../assets/images/CTY-logo-rectangle.png";
+
+const GrantAccessForm = () => {
+	const [role, setRole] = useState("bidder");
+	const [buyerDetails, setBuyerDetails] = useState({
+		userType: "buyer",
+		company: "",
+		description: "",
+		address: "",
+		email: "",
+	});
+
+	const [bidderDetails, setBidderDetails] = useState({
+		userType: "bidder",
+		firstName: "",
+		lastName: "",
+		email: "",
+	});
+
+	const [resgisterStatus, setRegisterStatus] = useState("");
+	const [validationErrors, setValidationErrors] = useState([]);
+
+	async function postDetails(endpoint, data) {
+		try {
+			const res = await post(endpoint, data);
+
+			if (res && res.code !== "VALIDATION_ERROR") {
+				setRegisterStatus("Successfully registered.");
+				setValidationErrors([]);
+			} else {
+				setRegisterStatus("Registration failed!");
+				setValidationErrors(res.errors || []);
+			}
```

Comment:
If the catch block is executed whenever we get an error, it means that the `if` is completely unnecessary

```suggestion
				setRegisterStatus("Successfully registered.");
				setValidationErrors([]);
```